scripts/cromwell/analyze_resource_acquisition.py

```diff
@@ -0,0 +1,460 @@
+	"""
+	# get start (start time of VM start) & end time (end time of 'ok') according to metadata
+	start = None
+	end = None
+
+	if 'executionEvents' in call_info:
+		for x in call_info['executionEvents']:
+			# ignore incomplete executionEvents (could be due to server restart or similar)
+			if 'description' not in x:
+				continue
+			y = x['description']
+
+			if 'backend' in call_info and call_info['backend'] == 'PAPIv2':
+				if y.startswith("PreparingJob"):
+					start = dateutil.parser.parse(x['startTime'])
+				if y.startswith("Worker released"):
+					end = dateutil.parser.parse(x['endTime'])
+			else:
+				if y.startswith("start"):
+					start = dateutil.parser.parse(x['startTime'])
+				if y.startswith("ok"):
+					end = dateutil.parser.parse(x['endTime'])
+
+	# if we are preempted or if cromwell used previously cached results, we don't even get a start time from JES.
+	# if cromwell was restarted, the start time from JES might not have been written to the metadata.
+	# in either case, use the Cromwell start time which is earlier but not wrong.
+	if start is None:
+		start = dateutil.parser.parse(call_info['start'])
+
+	# if we are preempted or if cromwell used previously cached results, we don't get an endtime from JES right now.
+	# if cromwell was restarted, the start time from JES might not have been written to the metadata.
+	# in either case, use the Cromwell end time which is later but not wrong
+	if end is None:
+		end = dateutil.parser.parse(call_info['end'])
+
+	return start, end
+
+def get_mem_cpu(m):
+	"""
+	Modified from: https://github.com/broadinstitute/dsde-pipelines/blob/develop/scripts/calculate_cost.py
+	"""
+	cpu = 'na'
+	memory = 'na'
+	if 'runtimeAttributes' in m:
+		if 'cpu' in m['runtimeAttributes']:
+			cpu = int(m['runtimeAttributes']['cpu'])
+		if 'memory' in m['runtimeAttributes']:
+			mem_str = m['runtimeAttributes']['memory']
+			memory = float(mem_str[:mem_str.index(" ")])
+	return cpu, memory
+
+def getCalls(m, alias=None):
+	"""
+	Modified from download_monitoring_logs.py script by Mark Walker
+	https://github.com/broadinstitute/gatk-sv/blob/master/scripts/cromwell/download_monitoring_logs.py
+	"""
+	if isinstance(m, list):
+		call_metadata = []
+		for m_shard in m:
+			call_metadata.extend(getCalls(m_shard, alias=alias))
+		return call_metadata
+
+	if 'labels' in m:
+		# In alias, track hierarchy of workflow/task up to current task nicely without repetition
+		if alias is None:
+			alias = ""
+		to_add = ""
+		if 'wdl-call-alias' in m['labels']:
+			to_add = m['labels']['wdl-call-alias']
+		elif 'wdl-task-name' in m['labels']:
+			to_add = m['labels']['wdl-task-name']
+		if to_add != "" and not alias.endswith(to_add):
+			if alias != "" and alias[-1] != ".":
+				alias += "."
+			alias += to_add
+
+	
+
+	call_metadata = []
+	if 'calls' in m:
+		for call in m['calls']:
+			# Skips scatters that don't contain calls
+			if '.' not in call:
+				continue
```

Comment:
How many indents is too many indents? I think the answer is 4. You could consider breaking some of this function into multiple helper functions to reduce indents and improve readability.

Reply:
Very fair point, although this particular line is copied from your script :) I created helper functions for most of the places where there were too many indents, but it was not straightforward to do so for this line or several in `calculate_start_end()` (which also originated in a different script, possibly why it isn't as easy to fix). I'll come back to them if necessary, but it is greatly reduced now!